lib/archive/unpack.go

```diff
@@ -45,7 +45,14 @@ const (
 func Unpack(op trace.Operation, tarStream io.Reader, filter *FilterSpec, root string) error {
 	op.Debugf("unpacking archive to root: %s, filter: %+v", root, filter)
 
-	tr := tar.NewReader(tarStream)
+	// Online datasource is sending a tar reader instead of an io reader.
+	// Type check here to see if we actually need to create a tar reader.
+	var tr *tar.Reader
+	if trCheck, ok := tarStream.(*tar.Reader); ok {
```

Comment:
I meant is there a situation wherein we wouldn't be passed a `tar.Reader`? Or, alternatively, should we stick to the convention of expecting an `io.ReadCloser`? Just curious, not a show-stopper.